core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -52,6 +66,23 @@ abstract class SparkFunSuite
     getTestResourceFile(file).getCanonicalPath
   }
 
+  private def runningThreadNames(): Set[String] = {
+    Thread.getAllStackTraces.keySet().map(_.getName).toSet
+  }
+
+  private def printRemainingThreadNames(): Unit = {
+    val currentThreadNames = runningThreadNames()
+    val whitelistedThreadNames = currentThreadNames
+      .filterNot { s => threadWhiteList.exists(s.matches(_)) }
+    val remainingThreadNames = whitelistedThreadNames.diff(beforeAllTestThreadNames)
```

Comment:
nit: I think this would be better written as:

  val remainingThreadNames = runningThreadNames.diff(beforeAllTestThreadNames).filterNot { s => threadWhiteList.exists(s.matches(_)) }

(although putting the whitelist filtering into runningThreadNames() would still make this more concise).

The reason is that it's not obvious to the reader why you whitelist 'after' threads but not 'before' - clearer to whitelist the diff.

Reply:
Compressed.